mono/metadata/reflection.c

```diff
@@ -2196,7 +2196,10 @@ MonoType*
 mono_reflection_type_from_name (char *name, MonoImage *image)
 {
 	MonoError error;
-	MonoType  *result = mono_reflection_type_from_name_checked (name, image, &error);
+	error_init (&error);
```

Comment:
Unconditional initialization needed due to a cleanup -- a mess.

Reply:
I guess could defer to leaf function but my newest understanding, in the presence of a cleanup function, is that this is a mess and better done this way, and it interoperates ok with existing code. There might be extra initializations, but in fact there already were. This pattern will actually be more maintainable and more efficient in the long run (C++ exceptions are even more efficient because the "peanut butter" initialization and constant checks both go away..)